status/status_test.go

```diff
@@ -18,43 +18,410 @@ package status_test
+					"pac.test.appstudio.openshift.io/url-repository": "devfile-sample-go-basic",
+					"pac.test.appstudio.openshift.io/sha":            "12a4a35ccd08194595179815e4646c3a6c08bb77",
+					"pac.test.appstudio.openshift.io/event-type":     "pull_request",
+				},
+				Annotations: map[string]string{
+					"build.appstudio.redhat.com/commit_sha":         "6c65b2fcaea3e1a0a92476c8b5dc89e92a85f025",
+					"appstudio.redhat.com/updateComponentOnSuccess": "false",
+					"pac.test.appstudio.openshift.io/repo-url":      "https://github.com/devfile-sample/devfile-sample-go-basic",
+				},
+			},
+			Spec: applicationapiv1alpha1.SnapshotSpec{
+				Application: "application-sample",
+				Components: []applicationapiv1alpha1.SnapshotComponent{
+					{
+						Name:           "component-sample",
+						ContainerImage: "sample_image",
+						Source: applicationapiv1alpha1.ComponentSource{
+							ComponentSourceUnion: applicationapiv1alpha1.ComponentSourceUnion{
+								GitSource: &applicationapiv1alpha1.GitSource{
+									Revision: "sample_revision",
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		mockK8sClient = &MockK8sClient{
+			getInterceptor: func(key client.ObjectKey, obj client.Object) {
+				if taskRun, ok := obj.(*tektonv1.TaskRun); ok {
+					if key.Name == successfulTaskRun.Name {
+						taskRun.Status = successfulTaskRun.Status
+					} else if key.Name == failedTaskRun.Name {
+						taskRun.Status = failedTaskRun.Status
+					} else if key.Name == skippedTaskRun.Name {
+						taskRun.Status = skippedTaskRun.Status
+					}
+				}
+				if plr, ok := obj.(*tektonv1.PipelineRun); ok {
+					if key.Name == pipelineRun.Name {
+						plr.Status = pipelineRun.Status
+					}
+				}
+			},
+			listInterceptor: func(list client.ObjectList) {},
+		}
+
+		githubSnapshot = &applicationapiv1alpha1.Snapshot{
 			ObjectMeta: metav1.ObjectMeta{
 				Labels: map[string]string{
 					"pac.test.appstudio.openshift.io/git-provider": "github",
 				},
 			},
 		}
+
+		ctrl := gomock.NewController(GinkgoT())
+		mockReporter = status.NewMockReporterInterface(ctrl)
+		mockReporter.EXPECT().GetReporterName().Return("mocked-reporter").AnyTimes()
+	})
+
+	It("can get reporters from a snapshot", func() {
+		st := status.NewStatus(logr.Discard(), nil)
+		reporter := st.GetReporter(githubSnapshot)
+		Expect(reporter).ToNot(BeNil())
+		Expect(reporter.GetReporterName()).To(Equal("GithubReporter"))
+	})
+
+	It("doesn't report anything when there are not test results", func() {
+
+		mockReporter.EXPECT().Initialize(gomock.Any(), gomock.Any()).Times(0)   // without test results reporter shouldn't be initialized
+		mockReporter.EXPECT().ReportStatus(gomock.Any(), gomock.Any()).Times(0) // without test results reported shouldn't report status
+
+		st := status.NewStatus(logr.Discard(), nil)
+		err := st.ReportSnapshotStatus(context.Background(), mockReporter, githubSnapshot)
+		Expect(err).To(Succeed())
```

Comment:
non-blocking suggestion: While `Succeed` works here just fine, I think using `Expect(err).NotTo(HaveOccurred())` is easier to read when we're checking an err variable.

Reply:
updated

Reply:
I'm still seeing it on this line, but can see it updated on other lines.

Reply:
updated everywhere

Reply:
Thanks!